azuki-core/src/main/kotlin/com/anaplan/engineering/azuki/core/runner/MultiOracleScenarioRunner.kt

```diff
@@ -41,52 +42,44 @@ class MultiOracleScenarioRunner<
             Log.info("Verification result, oracle=$oracle verified=${oracleResult.verified}")
             oracleResults.add(oracleResult)
         }
-        return Result(oracleResults)
+        return Result(testInstance, oracleResults)
     }
 
-    private fun verifyWithOracle(oracleInstance: ImplementationInstance<AF, CF, QF, AGF>): OracleResult<AF, QF, AGF> {
+    private fun verifyWithOracle(oracleInstance: ImplementationInstance<AF, CF, QF, AGF>): OracleResult<AF, CF, QF, AGF> {
+        val resultBuilder = OracleResult.Builder(oracleInstance)
         val declarationValidTaskResult = isDeclarationValid(oracleInstance)
+        resultBuilder.add(declarationValidTaskResult)
         if (declarationValidTaskResult.result != true) {
-            return OracleResult(false,
-                oracleInstance.instanceName,
-                oracleInstance.version,
-                listOf(declarationValidTaskResult))
+            return resultBuilder.build()
         }
         val buildActionsValidTaskResult = areBuildActionsValid(oracleInstance)
+        resultBuilder.add(buildActionsValidTaskResult)
         if (buildActionsValidTaskResult.result != true) {
-            return OracleResult(false,
-                oracleInstance.instanceName, oracleInstance.version,
-                listOf(declarationValidTaskResult, buildActionsValidTaskResult))
+            return resultBuilder.build()
         }
-        // TODO - c/should we cache generate/query of test system
+        // TODO - must retain generate of test system -- or we get different scenario!
```

Comment:
I'm not entirely sure what this means. Might be worth rephrasing the comment.

Reply:
doesn't really matter -- I need todo it before I finish :rofl:

Reply:
basically means that if that we can't do what we're doing now or we have different generated actions for each oracle!